errcheck/errcheck.go

```diff
@@ -586,18 +586,29 @@ func (v *visitor) Visit(node ast.Node) ast.Visitor {
 			for _, name := range vspec.Names {
 				lhs = append(lhs, ast.Expr(name))
 			}
-			v.checkAssignment(lhs, vspec.Values)
+			followed := v.checkAssignment(lhs, vspec.Values)
+			if !followed {
+				return nil
+			}
 		}
 
 	case *ast.AssignStmt:
-		v.checkAssignment(stmt.Lhs, stmt.Rhs)
+		followed := v.checkAssignment(stmt.Lhs, stmt.Rhs)
+		if !followed {
+			return nil
+		}
+
+	case *ast.TypeAssertExpr:
+		v.checkAssertExpr(stmt)
+		return nil
 
 	default:
 	}
 	return v
 }
 
-func (v *visitor) checkAssignment(lhs, rhs []ast.Expr) {
+func (v *visitor) checkAssignment(lhs, rhs []ast.Expr) (followed bool) {
+	followed = true
```

Comment:
You should probably convert everything here to explicit returns (seems like only line 616 and 619 left) so that there's one way to do things.

Please add a doscstring explaining what return value means here.